lib/engine/step/g_18_al/bye_sell_par_shares.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_relative '../buy_sell_par_shares'
+
+module Engine
+  module Step
+    module G18AL
+      class BuySellParShares < BuySellParShares
+        def setup
+          super
+          @controller_at_start = current_controller
+        end
+
+        def log_pass(entity)
+          super
+
+          return if @controller_at_start.nil? || @controller_at_start == current_controller
+
+          @log << "#{route_bonus_ability.owner.name} removes route bonuses as presidency changed"
+          president_changed(route_bonus_ability.owner)
+          @controller_at_start = nil
+        end
+
+        private
+
+        def current_controller
+          route_bonus_ability&.owner&.owner
```

Comment:
i know i told you to do this but there's a better method, route_bonus_ability&.player

Reply:
Done